### PR TITLE
Use OkHttp instead of java.net.

### PR DESCRIPTION
**Pull request by kuiperanon**
RE: https://github.com/tcheeric/nostr-java/issues/161

```
        return this.scheme.replaceFirst("ws", "http") + "://" + this.host;
```

This part is a bit hacky, but it's required because OkHttp will only accept URIs that have the "http" or "https" schemes. It still works like this, seemingly.

### CODE REVIEW

**Comment by kuiperanon** [comment]
I have noticed that I get strange different results when running this on Android. I'll try to set up a minimal example to see if I reproduce the issue.

**Comment by kuiperanon** [comment]
IMPORTANT. I disabled some tests on:

`nostr-java-test/src/test/java/nostr/test/client/ClientTest.java`

Please don't accept this PR until I re-enable them and figure out why they weren't passing for me.

File: nostr-java-event/src/main/java/nostr/event/BaseTag.java [hunk 1]
```diff
@@ -15,8 +15,8 @@
 import nostr.event.json.serializer.TagSerializer;
 import nostr.util.NostrException;
 
-import java.beans.IntrospectionException;
-import java.beans.PropertyDescriptor;
+import com.googlecode.openbeans.IntrospectionException;
+import com.googlecode.openbeans.PropertyDescriptor;
```
Comment (kuiperanon) [inline nostr-java-event/src/main/java/nostr/event/BaseTag.java:19]: These changes were necessary for this to compile for Android.

**Comment by tcheeric** [comment]
I am getting a few test failures after merging into the develop branch. I think it conflicts with Nick's changes. I am sure not a big deal, but can you please look into it when you have a chance? 
Thanks.

**Comment by kuiperanon** [comment]
@tcheeric I'll look into it tomorrow for sure!

**Comment by avlo** [comment]
awesome addition of Relay URI format validation- it caught that discrepancy in my unit test strings.  

unit tests updated and now successfully passing once again, changes submitted in [PR 168](https://github.com/tcheeric/nostr-java/pull/168)

**Comment by tcheeric** [comment]
Thank you! Will merge today. 
-- 
 Sent with Tuta; enjoy secure & ad-free emails: 
 https://tuta.com


27 May 2024, 05:15 by ***@***.***:

**Pull request merged**